uri/uri.cpp

```diff
@@ -97,4 +101,63 @@ Uri Uri::parse(std::string uristr, std::optional<std::reference_wrapper<Uri cons
     return uri;
 }
 
+// https://w3c.github.io/FileAPI/#unicodeBlobURL
+std::string blob_url_create(Origin *origin) {
```

Comment:
Why a mutable pointer? You don't change it, and the spec makes it sound mandatory, so `Origin const &`.